pyensembl/database.py

```diff
@@ -81,25 +82,42 @@ def _create_database(self):
             indices=indices)
         return db
 
-    def _connect_or_create_database(self):
+    def connection_if_exists(self):
         """
-        If database already exists, open a connection.
-        Otherwise, create it.
+        Return the connection if the DB exists, and otherwise return 
+        None.
         """
         db_path = self.local_db_path()
         if exists(db_path):
-            db = sqlite3.connect(db_path)
+            connection = sqlite3.connect(db_path)
             # maybe file got created but not filled
-            if datacache.db.db_table_exists(db, 'ensembl'):
-                return db
-        return self._create_database()
+            if datacache.db.db_table_exists(connection, 'ensembl'):
+                return connection
+        return None
+
+    def _connect_or_create_database(self):
+        """
+        If database already exists, open a connection.
+        Otherwise, create it if auto download is enabled.
+        """
+        connection = self.connection_if_exists()
+        if connection:
+            return connection
+        if self.auto_download:
```

Comment:
Good idea. What about situations where we want to force a SQLite setup without re-downloading? That's probably overcomplicating things.